src/org/thoughtcrime/securesms/database/ThreadDatabase.java

```diff
@@ -150,11 +157,34 @@ private void updateThread(long threadId, long count, String body, @Nullable Uri
       contentValues.put(ARCHIVED, 0);
     }
 
-    SQLiteDatabase db = databaseHelper.getWritableDatabase();
-    db.update(TABLE_NAME, contentValues, ID + " = ?", new String[] {threadId + ""});
-    notifyConversationListListeners();
+    if (!messageCache.containsKey(threadId)) {
+      messageCache.put(threadId, new ArrayList<>());
+    }
+    messageCache.get(threadId).add(contentValues);
+
+    if (!recipientCache.containsKey(threadId)) {
+      recipientCache.put(threadId, getRecipientForThreadId(threadId).getName().replaceAll("\\s*", ""));
```

Comment:
```suggestion
      Recipient recipient = getRecipientForThreadId(threadId);
      if (recipient != null) {
        recipientCache.put(threadId, recipient.getAddress().serialize());
      }
```

Reply:
Also it might be better if we put the `recipientCache` inside `getRecipientForThreadId` function instead so the whole app can benefit